services/slam/fake/slam.go

```diff
@@ -31,12 +30,47 @@ import (
 
 var model = resource.NewDefaultModel("fake")
 
+const datasetDirectory = "slam/example_cartographer_outputs/viam-office-02-22-1"
+
+type pose struct {
+	X float64 `json:"x"`
+	Y float64 `json:"y"`
+	Z float64 `json:"z"`
+}
+
+type poseInFrame struct {
+	Pose pose `json:"pose"`
+}
+
+type quat struct {
+	Imag float64 `json:"imag"`
+	Jmag float64 `json:"jmag"`
+	Kmag float64 `json:"kmag"`
+	Real float64 `json:"real"`
+}
+
+type extra struct {
+	Quat quat `json:"quat"`
+}
+
+type position struct {
+	Pose  poseInFrame `json:"pose"`
+	Extra extra       `json:"extra"`
+}
+
+// type positionNew struct {
+// 	Pose               pose   `json:"pose"`
+// 	ComponentReference string `json:"component_reference"`
+// 	Extra              extra  `json:"extra"`
+// }
+
 const (
-	internalStateTemplate = "slam/example_cartographer_outputs/internal_state/internal_state_%d.pbstream"
+	internalStateTemplate = datasetDirectory + "/internal_state/internal_state_%d.pbstream"
 	maxDataCount          = 16
-	pcdTemplate           = "slam/example_cartographer_outputs/pointcloud/pointcloud_%d.pcd"
```

Comment:
@jeremyrhyde I'm also inclined to delete the artifact dataset you uploaded given that:
1. The images are collected as PNGs rather than jpegs
2. The position data is collected as a custom text format which requires custom parsing rather than json & I'd rather not keep around the custom parsing code in fake and instead require any future fake slam datasets to collect structured data as the json.

I think any fake slam datasets we upload to artifact should be reproducible by anyone on the SLAM team with minimal setup i.e. w/o needing to cherry-pick modifications to RDK. 

What do you think?

Reply:
done